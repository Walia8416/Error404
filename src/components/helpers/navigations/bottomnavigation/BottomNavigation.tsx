import React from 'react';
import {View, Text, Image} from 'react-native';
import {RFValue} from 'react-native-responsive-fontsize';
import {Colors} from '../../../../constants/colors';
import {heighttodp, widthtodp} from '../../../../constants/Dimenstions';
import {Medium, Regular} from '../../../../constants/Fonts';
import Images from '../../../../constants/icon';
import ImageContainer from '../../header/ImageContainer';

const images = [
  Images.home,
  Images.home,
  Images.crop,
  Images.crop,
  Images.weather,
  Images.weather,
  Images.store,

  Images.store,
  Images.learngray,
  Images.shorts,
  Images.shortsgray,
];

const BottomNavigation = ({
  index,
  name,
  active,
}: {
  index: number;
  name?: string;
  active?: boolean;
}) => {
  images[index];

  return (
    <View style={{justifyContent: 'center', alignItems: 'center'}}>
      <View
        style={{
          height: heighttodp(24),
          width: widthtodp(24),
          justifyContent: 'center',
          alignItems: 'center',
          marginBottom: 5,
        }}>
        <ImageContainer
          name={images[index]}
          style={{width: widthtodp(35), height: heighttodp(35)}}
        />
      </View>
      {name && (
        <View
          style={{
            width: widthtodp(80),
            alignItems: 'center',
          }}>
          <Text
            numberOfLines={1}
            style={{
              fontFamily: active ? Medium : Regular,
              fontWeight: active ? '500' : '400',
              color: active ? '#ffde59' : Colors.FeatherWhite,
              fontSize: RFValue(8),
            }}>
            {name}
          </Text>
        </View>
      )}
    </View>
  );
};

export default BottomNavigation;
