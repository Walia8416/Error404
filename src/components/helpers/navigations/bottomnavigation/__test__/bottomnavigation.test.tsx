import { act, create } from 'react-test-renderer';

import BottomNavigation from '../BottomNavigation';
import React from 'react';

describe('bottomNavigation', () => {
  it('render test', () => {
    const bottom = create(<BottomNavigation index={0} />);
    expect(bottom).toMatchSnapshot()
  });
});
