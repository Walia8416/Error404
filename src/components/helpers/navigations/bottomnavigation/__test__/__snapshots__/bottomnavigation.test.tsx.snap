// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bottomNavigation render test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": 50,
        "justifyContent": "center",
        "width": 50,
      }
    }
  >
    <Image
      height={50}
      resizeMode="contain"
      source={
        Object {
          "process": [Function],
        }
      }
      width={50}
    />
  </View>
</View>
`;
